bindings_wasm/src/mls_client.rs

```diff
@@ -1,41 +1,54 @@
 use js_sys::Uint8Array;
 use std::collections::HashMap;
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
 use wasm_bindgen::prelude::{wasm_bindgen, JsError};
 use wasm_bindgen::JsValue;
 use xmtp_api_http::XmtpHttpApiClient;
 use xmtp_cryptography::signature::ed25519_public_key_to_address;
-use xmtp_id::associations::{
-  generate_inbox_id as xmtp_id_generate_inbox_id, unverified::UnverifiedSignature, AccountId,
-  MemberIdentifier,
-};
-use xmtp_mls::api::ApiClientWrapper;
+use xmtp_id::associations::builder::SignatureRequest;
+use xmtp_id::associations::unverified::UnverifiedSignature;
 use xmtp_mls::builder::ClientBuilder;
 use xmtp_mls::identity::IdentityStrategy;
-use xmtp_mls::retry::Retry;
+use xmtp_mls::storage::consent_record::StoredConsentRecord;
 use xmtp_mls::storage::{EncryptedMessageStore, EncryptionKey, StorageOption};
 use xmtp_mls::Client as MlsClient;
 
+use crate::consent_state::{WasmConsent, WasmConsentEntityType, WasmConsentState};
+use crate::conversations::WasmConversations;
+use crate::inbox_state::WasmInboxState;
+
 pub type RustXmtpClient = MlsClient<XmtpHttpApiClient>;
 
+#[wasm_bindgen]
+#[derive(Clone, Eq, Hash, PartialEq)]
+pub enum WasmSignatureRequestType {
+  AddWallet,
+  CreateInbox,
+  RevokeWallet,
+  RevokeInstallations,
+}
+
 #[wasm_bindgen]
 pub struct WasmClient {
   account_address: String,
   inner_client: Arc<RustXmtpClient>,
-  signatures: HashMap<MemberIdentifier, UnverifiedSignature>,
+  signature_requests: Arc<Mutex<HashMap<WasmSignatureRequestType, SignatureRequest>>>,
 }
 
-#[wasm_bindgen]
+#[wasm_bindgen(js_name = createClient)]
 pub async fn create_client(
   host: String,
   inbox_id: String,
   account_address: String,
+  db_path: String,
   encryption_key: Option<Uint8Array>,
   history_sync_url: Option<String>,
 ) -> Result<WasmClient, JsError> {
+  xmtp_mls::utils::wasm::init().await;
```

Comment:
we shouldn't include the tracing setup in prod, instead we should come up with some system similar to the `bindings_ffi` to pass a logger from javascript/let the client sdk setup logging by calling the init fn themselves. Could be a good followup issue since it's not necessary yet